geomstats/geometry/symmetric_matrices.py

```diff
@@ -133,3 +135,208 @@ def matrix_representation(basis_representation):
+        -------
+        submersed_point : array-like, shape=[..., n]
+        """
+        return Matrices.diagonal(point)
+
+    def tangent_submersion(self, vector, point):
+        """Tangent submersion.
+
+        Parameters
+        ----------
+        vector : array-like, shape=[..., n, n]
+        point : Ignored.
+
+        Returns
+        -------
+        submersed_vector : array-like, shape=[..., n]
+        """
+        out = self.submersion(vector)
+        return repeat_out(self.point_ndim, out, vector, point, out_shape=(self.n,))
+
+    def _create_basis(self):
+        """Compute the basis of the vector space of hollow symmetric matrices."""
+        indices, values = [], []
+        k = -1
+        for row in range(self.n):
+            for col in range(row + 1, self.n):
+                k += 1
+                indices.extend([(k, row, col), (k, col, row)])
+                values.extend([1.0, 1.0])
+
+        return gs.array_from_sparse(indices, values, (k + 1, self.n, self.n))
+
+    @staticmethod
+    def basis_representation(matrix_representation):
+        """Convert a hollow symmetric matrix into a vector.
+
+        Parameters
+        ----------
+        matrix_representation : array-like, shape=[..., n, n]
+            Matrix.
+
+        Returns
+        -------
+        vec : array-like, shape=[..., n(n+1)/2]
+            Vector.
+        """
+        return gs.triu_to_vec(matrix_representation, k=1)
+
+    def projection(self, point):
+        """Project a point in embedding manifold on embedded manifold.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., *embedding_space.point_shape]
+            Point in embedding manifold.
+
+        Returns
+        -------
+        projected : array-like, shape=[..., *point_shape]
+            Projected point.
+        """
+        return point - Matrices.to_diagonal(point)
+
+
+class HollowMatricesPermutationInvariantMetric(FlatRiemannianMetric):
+    r"""A permutation-invariant metric on the space of hollow matrices.
+
+    It is flat Riemannian metric invariant by the congruence action
+    of permutation matrices defined over a matrix vector space.
+
+    Its associated quadratic form is:
+
+    .. math::
+
+        (X)=\alpha \operatorname{tr}\left(X^2\right)
+        +\beta \operatorname{Sum}\left(X^2\right)
+        +\gamma \operatorname{Sum}(X)^2
+
+    Check out chapter 8 of [T2022]_ for more details.
+
+    References
+    ----------
+    .. [T2022] Yann Thanwerdas. Riemannian and stratified
+    geometries on covariance and correlation matrices. Differential
+    Geometry [math.DG]. Université Côte d'Azur, 2022.
+    """
+
+    def __init__(self, space, alpha=1.0, beta=1.0, gamma=1.0):
+        self._check_params(space, alpha, beta, gamma)
+        super().__init__(space=space)
+        self.alpha = alpha
+        self.beta = beta
+        self.gamma = gamma
+
+    @staticmethod
+    def _check_params(space, alpha, beta, gamma):
```

Comment:
Thx for the eqn, super helpful.

Need parameter descriptions (even though, I agree, it seems clear if we scroll up)